src/LanguageServer/ProtocolUnitTests/Microsoft.CodeAnalysis.LanguageServer.Protocol.UnitTests.csproj

```diff
@@ -23,6 +23,7 @@
     <ProjectReference Include="..\..\Compilers\Core\Portable\Microsoft.CodeAnalysis.csproj" />
     <ProjectReference Include="..\..\Compilers\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.csproj" />
     <ProjectReference Include="..\..\Compilers\VisualBasic\Portable\Microsoft.CodeAnalysis.VisualBasic.vbproj" />
+    <ProjectReference Include="..\..\EditorFeatures\Core\Microsoft.CodeAnalysis.EditorFeatures.csproj" />
```

Comment:
@dibarbet / @tmat / @CyrusNajmabadi is this a layering violation? Language Server is in "features", but not sure if it matters since it's just tests.

Reply:
It already existed. You can see I fixed some of the violation code in `src/LanguageServer/ProtocolUnitTests/Completion/CompletionResolveTests.cs` that was referencing `Microsoft.VisualStudio.Text.Adornments`

I don't immediately remember how this was included transitively before but I can get a binlog from main and look

Reply:
It was probably `Microsoft.CodeAnalysis.Remote.ServiceHub.csproj` referencing `Microsoft.CodeAnalysis.ExternalAccess.Razor.csproj` which references `Microsoft.CodeAnalysis.EditorFeatures.csproj`

I'm fine with this then as it is a pre-existing violation then

Reply:
ah yea that's definitely it